Scripts/ImageFactory/Scripts/DistributionHelpers.psm1

```diff
@@ -1,8 +1,8 @@
 ﻿function SelectSubscription($subId){
     # switch to another subscription assuming it's not the one we're already on
-    if((Get-AzureRmContext).Subscription.SubscriptionId -ne $subId){
+    if((Get-AzureRmContext).Subscription.Id -ne $subId){
         Write-Output "Switching to subscription $subId"
-        Select-AzureRmSubscription -SubscriptionId $subId | Out-Null
+        Set-AzureRmContext -SubscriptionId $subId | Out-Null
```

Comment:
This is affected by this bug:  https://github.com/Azure/azure-powershell/issues/3954

A workaround is to remove the subscription check in SelectSubscription.